doc/kubernetes/modules/ROOT/pages/running/keycloak-with-external-infinispan-deployment.adoc

```diff
@@ -0,0 +1,62 @@
+= Keycloak with external Infinispan Deployment
+:description: This describes configuration tweaks and changes on how to run Keycloak with an external Infinispan Deployment under load.
+
+{description}
+
+== Audience
+
+This guide describes advanced Infinispan configurations for Keycloak on Kubernetes which are load tested and will recover from single Pod failures.
+
+While the Helm charts in the Keycloak Benchmark project mix different aspects of production style deployments with instrumentation and monitoring, this documentation focuses on a minimal deployment with optional add-ons which admins can opt in for their own deployments.
+
+See xref:running/index.adoc[] for additional guides.
+
+== Prerequisites
+
+* OpenShift or Kubernetes cluster running.
+* Understanding of the xref:running/keycloak-deployment.adoc[Basic Keycloak deployment].
+* Understanding of an https://infinispan.org/docs/infinispan-operator/main/operator.html[Infinispan deployment using its Operator].
+
+== Procedure
+
+* Prepare an Infinispan Cache configuration XML, which would then be consumed in the Keycloak CR.
+* Create Config Map, Secret and other Kubernetes resources needed for Keycloak to connect to an external Infinispan deployment.
+* Extend the Keycloak Custom Resource with below Kubernetes deployment resources.
+
+=== Remote store username and password defined in a Kubernetes Secret
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: remote-store-secret
+  namespace: keycloak
+type: Opaque
+data:
+  username: <base-64 encoded String>
+  password: <base-64 encoded String>
+----
+
+=== Keycloak Cache configuration defined in a Kubernetes Config Map
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kcb-infinispan-cache-config
+  namespace: keycloak
+data: {}
+----
+
+=== Keycloak CR with external Infinispan updates
+[source,yaml]
+----
+include::example$helm/keycloak-ispn.yaml[tag=keycloak-ispn]
+----
+<1> Custom cache configuration XML file definition, which includes configuration for remote or embedded Infinispan store.
+See xref:./concepts/infinispan-cache-config-xml.adoc[Infinispan remote cache configuration] for details.
+<2> The hostname/IP and the port of the remote cache Infinispan cluster.
+<3> The credentials required, username and password, to access the remote cache Infinispan cluster.
+<4> Mounting the cache configuration Volume in Kubernetes.
+<5> `jboss.site.name` is an arbitrary Infinispan site name which Keycloak needs for a given Infinispan deployment
```

Comment:
from the comment it should be visible this value is used as the site name for the embedded infinispan. For example:
```suggestion
<5> `jboss.site.name` is an arbitrary Infinispan site name which Keycloak needs for its embedded Infinispan deployment; this site name is related only to embedded Infinsipan and does not need to match any value from the external Infinispan deployment
```